### DISCUSSION TITLE
How to change the assets URL origin added by `@vite` directive?

### DISCUSSION THREAD

**Discussion by iagobruno** [laravel/framework#43117] [Q&A] [answered]
I'm trying to deploy an app on Heroku that already uses the new Vite plugin but I'm getting this "https required" errors:

![image](https://user-images.githubusercontent.com/3616259/178137525-a5add4b9-54a1-4bbb-8374-66459ea30884.png)

Trying to solve the problem I noticed that the Vite manifest generated by the build command doesn't specify the origin or the protocol of the asset urls.

![image](https://user-images.githubusercontent.com/3616259/178137701-45df5974-68a1-4777-ad6b-f64d7fa02a61.png)

But the `@vite` directive automatically prepends the site origin in the asset urls.

![image](https://user-images.githubusercontent.com/3616259/178137881-44983e9b-61c9-4177-811b-c7b515001cb4.png)

Is there any way to change this origin or even disable it to prevent these errors and the assets being loaded?
I even looked [here in the repository](https://github.com/laravel/framework/blob/baf526ed6e725d7aa0950b0675361f4009a892af/src/Illuminate/Foundation/Vite.php) trying to find where is the logic that formats these urls but I didn't find anything.

Can anyone help me or guide me in some way?

**Comment by kaozaza2** [comment]
Set env `APP_URL` to `https://<full domain>`

**Reply by iagobruno** [reply]
It was the first thing I tried but it didn't work, the assets domain doesn't change if I change this environment variable.

**Comment by iagobruno** [comment, marked as answer]
## Solution

Recently the Laravel core team added an [env variable to customize the assets URL prefix](https://laravel.com/docs/9.x/vite#custom-base-urls).
So just configure this variable in your admin panel:

```
ASSET_URL=https://your-app.herokuapp.com
```

**Reply by abhijitbcob** [reply]
Thanks guys.

`ASSET_URL` this variable really saved me. our app has country prefix in the url.

**Reply by clemrep** [reply]
But if you have subdomain portal.your-app.com and your-app.com, what is the way to solve the problem ?
